src/malli/malli.cljc

```diff
@@ -0,0 +1,302 @@
+(ns malli.malli)
+
+(set! *warn-on-reflection* true)
+
+;;
+;; protocols
+;;
+
+(defprotocol IntoSchema
+  (-id [this])
+  (-aliases [this])
+  (-into-schema [this props childs opts]))
+
+(defprotocol Schema
+  (-validator [this opts])
+  (-props [this])
+  (-form [this]))
+
+(defn schema? [x]
+  (satisfies? Schema x))
+
+(defmethod print-method ::into-schema [v ^java.io.Writer w]
+  (.write w (str "#IntoSchema{:id " (-id v) "}")))
+
+(defmethod print-method ::schema [v ^java.io.Writer w]
+  (.write w (str (-form v))))
+
+;;
+;; impl
+;;
+
+(declare schema)
+(declare default-registry)
+
+(defn fail!
+  ([type]
+   (fail! type nil))
+  ([type data]
+   (throw (ex-info (str type) {:type type, :data data}))))
+
+(defn create-form [id props childs]
+  (cond
+    (and (seq props) (seq childs)) (into [id props] childs)
+    (seq props) [id props]
+    (seq childs) (into [id] childs)
+    :else id))
+
+(defn- -var-schema [v]
```

Comment:
Is this expecting that `v` is predicate function?

Reply:
removed.